docs/farming-&-staking/farming/additional-guides/network-info.md

```diff
@@ -6,72 +6,117 @@ slug: networking
-| 30433 | TCP      | Always public                                           | Autonomys networking P2P    | <ul><li>[Space Acres](../../../category/space-acres-recommended/)</li><li>[Advanced CLI Node (Consensus)](../../../category/advanced-cli)</li></ul> | Connecting to the rest of the network and retrieving archival history.            |
-| 30533 | TCP     | Always public                                           | Autonomys networking P2P    | <ul><li>[Advanced CLI Node (Consensus)](../../../category/advanced-cli)</li></ul>                                                                        | Connecting to the rest of the network and retrieving archival history, not required for Space Acres because the node and the farmer share the Autonomys networking stack there. |
-| 40333 | TCP          | Always public                                           | Autonomys networking P2P    | <ul><li>[Advanced CLI Node (Domain)](../../../category/operators-and-nominators)</li></ul>                                                                           | Connecting to the rest of the domain network for transaction gossipping.      |
-| 9944  | TCP (HTTP/WebSocket)| Public: <ul><li>RPC server</li></ul> Private: <ul><li>Internal node-to-farmer connections</li></ul> | Consensus RPC port         | <ul><li>[Advanced CLI Node (Consensus)](../../../category/advanced-cli)</li></ul>                                                                        | RPC server: Exposes various details about the blockchain (real-time and historical events, storage, transactions, etc) as well as submitting transactions to the network (balance transfers, staking, etc). <br /><br /> Farmer connections: The farmer connects in unsafe mode and receives notifications about slot challenges for block/vote production, archival history extension events, querying of various chain details. Nodes that process farmer connections should never be exposed publicly as public exposure of unsafe APIs can be used to not only prevent farmer from working, but even block node sync completely. |
-| 9945  | TCP (HTTP/WebSocket)| Public: <ul><li>RPC server</li></ul>                     | Domain RPC port            | <ul><li>[Advanced CLI Node (Domain)](../../../category/operators-and-nominators)</li></ul>                                                                           | RPC server: Expose various details about blockchain (real-time and historical events, storage, transactions, etc.) as well as submitting transactions to the network (balance transfers, interacting with smart contracts, etc). |
+| Port | Protocol | Component| Purpose |
+|--|--|--|--|
+| 30333  | TCP | Node (Consensus) | Facilitate block, transaction, and PoT gossip exchange. |
+| 30433  | TCP | Farmer | Retrieve archival history from the blockchain. |
+| 30533  | TCP | Farmer | Synchronize with node peers to access archival history. |
 
-### Special Note on Port 9944
+### Advanced CLI (Domain Operator)
```

Comment:
I don't think it's correct - the domain node uses the archival node data, so port 30533 is required to get the archival history, I'm not sure if the domain operator node needs access to 30333, but it likely does. 

@randywessels once again, not something I really tested myself, something you likely know better than I do

Reply:
This was a bit confusing for me as I had at one point several nodes running so I had to have unique ports for each.  I can confirm that 30333 and 30433 are needed for the node.  What I don't really know is about 30533.  I have it setup as a port trigger but I don't have a good way of seeing if it is being used.  We probably do need Nazar's input on that one.

As for 9944 and 9945, I am binding 9945 to my node container.  But it not forwarded.  And I have nothing for 9944.  But I think what I am really doing is using 9945 as how 9944 was intended if that makes sense.

I love the new format btw.  Much easier to understand.

Reply:
@nazar-pc can you help us out when you can, please?

Reply:
@nazar-pc Do you mind taking a look at this?  It was mainly done to clean up the page and make it easier to understand but we also want to double check if all of these ports are actually required and we are providing accurate information.  Specifically 30533 and 30433 we had questions about.

[New Page](https://deploy-preview-642--subspace-docs-staging.netlify.app/docs/farming-&-staking/farming/additional-guides/networking/)

[Existing Page](https://docs.autonomys.xyz/docs/farming-&-staking/farming/additional-guides/networking)

Reply:
RPC description for farming an public RPC purposes should be different, currently it says this:
> Exposes real-time and historical blockchain data, enables transaction submissions through HTTP and WebSocket, and is used for slot challenges.

Part of which only makes sense for farming purposes (and even then I would not talk about slot challenges because that is not the only thing it is used for) and the other part only makes sense for public RPC and is completely irrelevant for farming purposes.

The only thing that is completely incorrect is P2P port of the domain, it is actually `30334`.

Reply:
Changes proposed based on:
> RPC description for farming an public RPC purposes should be different, currently it says this:
> > Exposes real-time and historical blockchain data, enables transaction submissions through HTTP and WebSocket, and is used for slot challenges.

> Part of which only makes sense for farming purposes (and even then I would not talk about slot challenges because that is not the only thing it is used for) and the other part only makes sense for public RPC and is completely irrelevant for farming purposes.
- **Advanced CLI (Farmer)** Update the `Purpose` for 9944 to: `Enables transaction submissions through HTTP and WebSocket.`
- **RPC Server (Public)** Update the `Purpose` for 9944 to: `Allows retrieval of real-time and historical blockchain data.`

Changes proposed based on:
> The only thing that is completely incorrect is P2P port of the domain, it is actually 30334.
- **Advanced CLI (Domain Operator)** Remove `30433`, and add `30334`.

Would these changes be correct?

If I could also get some basic pointers as the the `Purpose` of the follow ports, I'd appreciate it.
- What would be a good updated `Purpose` be to describe port `9945` on a Public RPC (Domain) Server be? Something like `Allows retrieval of smart contract data.`?
- What would be a good `Purpose` to describe port `30334` on **Advanced CLI (Domain Operator)**?

Reply:
> * **Advanced CLI (Farmer)** Update the `Purpose` for 9944 to: `Enables transaction submissions through HTTP and WebSocket.`
> 
> * **RPC Server (Public)** Update the `Purpose` for 9944 to: `Allows retrieval of real-time and historical blockchain data.`

It is almost the opposite. For farmer it is sufficient to mention that it is needed for farmer operation, I don't see why more details are necessary. Farmer doesn't send any transactions there and doesn't use HTTP.

If node is used for RPC purposes then users will in fact send transactions to it and query all kinds of data from it (events, transactions, block contents, chain state, ...).

> **Advanced CLI (Domain Operator)** Remove `30433`, and add `30334`.

Don't touch `30433`. `40333` should be replaced with `30334`.

> What would be a good updated `Purpose` be to describe port `9945` on a Public RPC (Domain) Server be? Something like `Allows retrieval of smart contract data.`?

Similarly to node's RPC, but for domain: sending transactions (likely deploying or calling contracts, etc.). I don't really see the point of elaborating on it too much. Neither those who know what it is nor those who don't will benefit from it.

> What would be a good `Purpose` to describe port `30334` on **Advanced CLI (Domain Operator)**?

Just like for node, it is required for domain node to have good connectivity with P2P network.

---

Overall I would't elaborate on purpose too much. Here are more examples:

> Retrieve archival history from the blockchain.

Doesn't really retrieve history from the blockchain, it is used for both retrieving and serving pieces and metadata about archival history from P2P (farmers specifically).

> Synchronize with node peers to access archival history.

Technically farmers never reach out to nodes via P2P, nodes are only clients right now.